From e475653d17fb2012b537c2c37067795c0dd14ea8 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Thu, 28 Aug 2014 03:06:07 -0400
Subject: [acpi] scan: Clear match_driver flag in acpi_bus_trim()

Message-id: <20140828030607.2117.5606.stgit@gir.stowe>
Patchwork-id: 89313
O-Subject: [RHEL7.1 PATCH 25/30] ACPI / scan: Clear match_driver flag in acpi_bus_trim()
Bugzilla: 1128632
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1128632
Upstream Status: a951c773bc39677ef3fa42814be7f5218a3100b2

commit a951c773bc39677ef3fa42814be7f5218a3100b2
Author: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Date:   Mon Jan 27 23:08:09 2014 +0100

    ACPI / scan: Clear match_driver flag in acpi_bus_trim()

    Drivers should not bind to struct acpi_device objects that
    acpi_bus_trim() has been called for, so make that function
    clear flags.match_driver for those objects.

    If that is not done, an ACPI driver may theoretically try to operate
    a device that is not physically present.

    Fixes: 202317a573b2 (ACPI / scan: Add acpi_device objects for all device nodes in the namespace)
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
    Acked-by: Toshi Kani <toshi.kani@hp.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/acpi/scan.c b/drivers/acpi/scan.c
index 638d786..ba62d05 100644
--- a/drivers/acpi/scan.c
+++ b/drivers/acpi/scan.c
@@ -1989,6 +1989,7 @@ void acpi_bus_trim(struct acpi_device *adev)
  list_for_each_entry_reverse(child, &adev->children, node)
   acpi_bus_trim(child);
 
+ adev->flags.match_driver = false;
  if (handler) {
   if (handler->detach)
    handler->detach(adev);
-- 
1.7.1